integrated_channels/cornerstone/exporters/content_metadata.py

```diff
@@ -0,0 +1,108 @@
+    """
+    LONG_STRING_LIMIT = 10000
+
+    DATA_TRANSFORM_MAPPING = {
+        'ID': 'key',
+        'Title': 'title',
+        'Description': 'description',
+        'Thumbnail': 'card_image_url',
+        'URL': 'enrollment_url',
+        'IsActive': 'is_active',
+        'LastModifiedUTC': 'modified',
+        'Duration': 'estimated_hours',
+        'Owners': 'organizations',
+        'Languages': 'languages',
+        'Subjects': 'subjects',
+    }
+
+    def transform_organizations(self, content_metadata_item):
+        """
+        Return the transformed version of the course organizations
+        by converting each organization into cornerstone course owner object.
+        """
+        owners = []
+        for org in content_metadata_item.get('organizations', []):
+            owners.append({"Name": org})
+        return owners
+
+    def transform_is_active(self, content_metadata_item):
+        """
+        Return the transformed version of the course is_active status
+        by traversing course runs and setting IsActive to True if any of the course
+        runs have availability value set to `Current`, `Starting Soon` or `Upcoming`.
+        """
+        is_active = False
+        for course_run in content_metadata_item.get('course_runs', []):
+            if course_run.get('availability') in ['Current', 'Starting Soon', 'Upcoming']:
+                is_active = True
+                break
+        return is_active
+
+    def transform_modified(self, content_metadata_item):
+        """
+        Return the modified datetime of closest course run`.
+        """
+        modified_datetime = datetime.datetime.now(pytz.UTC)
+        course_runs = content_metadata_item.get('course_runs')
+        if course_runs:
+            closest_course_run = get_closest_course_run(course_runs)
+            modified_datetime = closest_course_run.get('modified', modified_datetime)
+
+        return str(modified_datetime)
+
+    def transform_estimated_hours(self, content_metadata_item):
+        """
+        Return the duration of course in hh:mm:ss format.
+        """
+        duration = None
+        course_runs = content_metadata_item.get('course_runs')
+        if course_runs:
+            closest_course_run = get_closest_course_run(course_runs)
+            estimated_hours = closest_course_run.get('estimated_hours')
+            if estimated_hours and isinstance(estimated_hours, (int, float)):
+                fraction, whole_number = math.modf(estimated_hours)
+                hours = "{:02d}".format(int(whole_number))
+                minutes = "{:02d}".format(int(60 * fraction))
+                duration = "{hours}:{minutes}:00".format(hours=hours, minutes=minutes)
+
+        return duration
+
+    def transform_languages(self, content_metadata_item):
+        """
+        Return the languages supported by course or `English` as default if no languages found.
+        """
+        return content_metadata_item.get('languages', ['English'])
```

Comment:
should this be `content_language`? I see we use that in the SAPSF exporter...

Reply:
`/api/v1/search/all/` API in discovery does not return `content_language` field. Wonder why we use it in SAPSF exporter.